client/src/services/todo.ts

```diff
@@ -75,42 +48,44 @@ export async function deleteCurrentTodo(id: string) {
 
 export async function fetchTodos(
   searchQuery: string,
-  filterBy: string,
+  filterBy: string | undefined,
   isAscending: boolean
 ) {
   try {
-    const token = checkForJwtToken()
-
-    const response = await axios.get(
-      `${
-        import.meta.env.VITE_API_BASE_URL
-      }/user/todo?query=${searchQuery}&filter_by=${filterBy}&is_ascending=${isAscending}`,
-      {
-        headers: {
-          Authorization: `Bearer ${token}`
-        }
-      }
-    )
-
+    const url = buildTodoURL(searchQuery, filterBy, isAscending)
+    const response = await axiosInstance.get(url)
     return response.data
   } catch (error) {
     console.error('Error fetching todos:', error)
     throw error
   }
 }
 
+function buildTodoURL(
```

Comment:
Almost perfect, just one thing, this function should return only the queryString so it could be reusable in other cases if needed.